src/fundus/scraping/pipeline.py

```diff
@@ -207,13 +213,25 @@ def crawl(
             only_unique=only_unique,
         )
 
-        event_loop = asyncio.new_event_loop()
-
-        while True:
-            try:
-                yield event_loop.run_until_complete(async_next(async_article_iter))
-            except StopAsyncIteration:
-                break
+        try:
+            event_loop = asyncio.get_running_loop()
+            if event_loop:
+                raise AssertionError(
+                    "There is already an eventloop running. If you want to crawl articles inside an "
+                    "async environment use crawl_async() instead."
+                )
+        except RuntimeError:
+            event_loop = asyncio.new_event_loop()
```

Comment:
Is there a case where the `if event_loop` check is false? From the asyncio documentation, we can omit the if-check. (Also eventloop -> event loop).

Also, I don't think this is an assertion error since the fault is not on our part but on how the user interacts with our API. In my opinion, throwing a runtime error fits this well. Implementing this is not entirely beautiful either with a boolean flag or intermediate exception. Here is my suggestion:

```suggestion
        try:
            asyncio.get_running_loop()
            raise AssertionError()
        except RuntimeError:
            event_loop = asyncio.new_event_loop()
        except AssertionError:
            raise RuntimeError(
                "There is already an event loop running. If you want to crawl articles inside an "
                "async environment use crawl_async() instead."
            )
```

Reply:
Ahh, nice idea to raise the exception with a different catch. Also thought `RuntimeError` would be the right one but hadn't an idea how to do it.